llms/mlx_lm/server.py

```diff
@@ -159,6 +161,40 @@ def do_POST(self):
         method = self.handle_stream if self.stream else self.handle_completion
         method(prompt, stop_id_sequences)
 
+    def validate_model_parameters(self):
+        """
+        Validate the model parameters passed in the request for the correct
+            types and values
+        """
+        if not isinstance(self.stream, bool):
+            raise ValueError("stream must be a boolean")
+        
+        if not isinstance(self.max_tokens, int) or self.max_tokens < 0:
+            raise ValueError("max_tokens must be a non-negative integer")
+        
+        if not isinstance(self.temperature, float) or self.temperature < 0:
+            raise ValueError("temperature must be a non-negative float")
+        
+        if not isinstance(self.top_p, float) or self.top_p < 0 or self.top_p > 1:
+            raise ValueError("top_p must be a float between 0 and 1")
+        
+        if not isinstance(self.repetition_penalty, float) or self.repetition_penalty < 0:
+            raise ValueError("repetition_penalty must be a non-negative float")
+        
+        if not isinstance(self.repetition_context_size, int) or self.repetition_context_size < 0:
+            raise ValueError("repetition_context_size must be a non-negative integer")
+        
+        if not isinstance(self.logit_bias, dict):
+            raise ValueError("logit_bias must be a dict of int to float")
+        
+        try:
+            self.logit_bias = {int(k): v for k, v in self.logit_bias.items()}
+        except ValueError:
+            raise ValueError("logit_bias must be a dict of int to float")
```

Comment:
What's the purpose of this? I don't think it makes sense to allow strings here does it?

Reply:
It's to support those who may call through postman:
<img width="984" alt="image" src="https://github.com/ml-explore/mlx-examples/assets/6634600/8571a3d0-a296-4a97-99ff-196593df27ec">